neps/nep-0245.md

```diff
@@ -382,24 +362,379 @@ Contracts which want to make use of `mt_transfer_call` and `mt_batch_transfer_ca
+
+Prior art:
+
+- NEAR's [Fungible Token Metadata Standard](../FungibleToken/Metadata.md)
+- NEAR's [Non-Fungible Token Metadata Standard](../NonFungibleToken/Metadata.md)
+- Discussion about NEAR's complete NFT standard: #171
+- Discussion about NEAR's complete Multi Token standard: #245
+
+### Metadata Interface
+
+Metadata applies at both the class level (`MTBaseTokenMetadata`) and the specific instance level (`MTTokenMetadata`). The relevant metadata for each:
+
+```ts
+
+type MTContractMetadata = {
+  spec: string, // required, essentially a version like "mt-1.0.0"
+  name: string, // required Zoink's Digitial Sword Collection
+}
+
+// Identify different FT or NFT category,
+// A combination of FT metadata and NFT ContractMetadata
+type MTBaseTokenMetadata = {
+  id: string, // required a contract-wise unique identifier for the metadata
+  name: string, // required, ex. "Silver Swords" or "Metaverse 3"
+  symbol: string|null, // required, ex. "MOCHI"
+  icon: string|null, // Data URL
+  decimals: string|null // should be not-null for FT and null for NFT,
+  base_uri: string|null, // Centralized gateway known to have reliable access to decentralized storage assets referenced by `reference` or `media` URLs
+  reference: string|null, // URL to a JSON file with more info
+  reference_hash: string|null, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+// Similar to TokenMetadata in NFT
+type MTTokenMetadata = {
+  title: string|null, // ex. "Arch Nemesis: Mail Carrier" or "Parcel #5055"
+  description: string|null, // free-form description
+  media: string|null, // URL to associated media, preferably to decentralized, content-addressed storage
+  media_hash: string|null, // Base64-encoded sha256 hash of content referenced by the `media` field. Required if `media` is included.
+  copies: number|null, // number of copies of this set of metadata in existence when token was minted.
+  issued_at: string|null, // When token was issued or minted, Unix epoch in milliseconds
+  expires_at: string|null, // When token expires, Unix epoch in milliseconds
+  starts_at: string|null, // When token starts being valid, Unix epoch in milliseconds
+  updated_at: string|null, // When token was last updated, Unix epoch in milliseconds
+  extra: string|null, // Anything extra the MT wants to store on-chain. Can be stringified JSON.
+  reference: string|null, // URL to an off-chain JSON file with more info.
+  reference_hash: string|null // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+// For base info, either base_id or base would be included, depends on implementation.
+// Suggest to return base to save client from extral query on base info.
+type MTTokenMetadataAll = {
+  base_id: string | null, // id in type MTBaseTokenMetadata
```

Comment:
This seems confusing, should this be instead restricted to one or the other (base_id vs including the metadata itself)?